Error when typing too fast (slow completion)

### Make sure you have done the following

- [x] Updated to the latest version of `blink.cmp`
- [x] Searched for existing issues and documentation (try `<C-k>` on https://cmp.saghen.dev)

### Bug Description

Here is the backtrace:

> Error executing vim.schedule lua callback: ...vim/plugged/blink.cmp/lua/blink/cmp/sources/lsp/init.lua:66: attempt to index local 'resolved_item' (a nil value)
> stack traceback:
>         ...vim/plugged/blink.cmp/lua/blink/cmp/sources/lsp/init.lua:66: in function 'handler'
>         .../neovim/0.10.2/share/nvim/runtime/lua/vim/lsp/client.lua:687: in function ''
>         vim/_editor.lua: in function <vim/_editor.lua:0>

### Relevant configuration

```lua
require("blink.cmp").setup {
  appearance = {
    use_nvim_cmp_as_default = true,
  },
  keymap = {
    preset = "super-tab",
    ["<C-e>"] = { "select_and_accept" },
    ["<C-d>"] = { "scroll_documentation_down", "fallback" },
    cmdline = {
      preset = "super-tab",
      ["<C-d>"] = { "show" },
    },
  },
  signature = { enabled = true },
  completion = {
    documentation = {
      auto_show = true,
      auto_show_delay_ms = 200,
      window = {
        border = "rounded",
      },
    },
  },
  sources = {
    default = { "lsp", "path", "snippets", "buffer", "ripgrep" },
    providers = {
      snippets = {
        score_offset = 3,
      },
      ripgrep = {
        name = "Ripgrep",
        module = "blink-ripgrep",
        score_offset = -3,
      },
    },
  },
}
```

### `neovim` version

NVIM v0.10.2

### `blink.cmp` version

main

I get this:

                                                                        {
  code = -32603,
  data = "java.util.concurrent.CompletionException: java.lang.IllegalStateException: Invalid completion proposal\n\tat java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)\n\tat java.base/java.util.concurrent.CompletableFuture.uniAcceptNow(CompletableFuture.java:747)\n\tat java.base/java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:735)\n\tat java.base/java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2214)\n\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:288)\n\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:201)\n\tat org.eclipse.jdt.ls.core.internal.ParentProcessWatcher.lambda$1(ParentProcessWatcher.java:144)\n\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:185)\n\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:97)\n\tat org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:114)\n\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\nCaused by: java.lang.IllegalStateException: Invalid completion proposal\n\tat org.eclipse.jdt.ls.core.internal.handlers.CompletionResolveHandler.resolve(CompletionResolveHandler.java:106)\n\tat org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.resolveCompletionItem(JDTLanguageServer.java:646)\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:580)\n\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$recursiveFindRpcMethods$0(GenericEndpoint.java:65)\n\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:128)\n\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:271)\n\t... 10 more\n",
  message = "Internal error.",
  <metatable> = {
    __tostring = <function 1>
  }
}
                                                                        "Internal error."

--- a/lua/blink/cmp/sources/lsp/init.lua
+++ b/lua/blink/cmp/sources/lsp/init.lua
@@ -60,7 +60,10 @@ function lsp:resolve(item, callback)
   item = require('blink.cmp.sources.lib.utils').blink_item_to_lsp_item(item)
 
   local success, request_id = client.request('completionItem/resolve', item, function(error, resolved_item)
-    if error or resolved_item == nil then callback(item) end
+    if error or resolved_item == nil then
+      callback(item)
+      return
+    end
 
     -- Snippet with no detail, fill in the detail with the snippet
     if resolved_item.detail == nil and resolved_item.insertTextFormat == vim.lsp.protocol.InsertTextFormat.Snippet then

